x-pack/plugin/esql/src/main/antlr/EsqlBaseLexer.g4

```diff
@@ -1,27 +1,52 @@
 lexer grammar EsqlBaseLexer;
 
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+@header {
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+}
+
+options {
+  superClass=LexerConfig;
+  caseInsensitive=true;
+}
```

Comment:
The main option that matters here is the superClass which is used to hook up the conditional code - the generated parser and lexer will invoke this code which is defined in the LexerConfig (for the lexer) and ParserConfig for the parser.